lib/krikri/job.rb

```diff
@@ -2,12 +2,32 @@ module Krikri
   ##
   # Generic Job class the gets extended by specific types of Jobs;
   # Harvest, Enrichment, etc.
+  #
+  # A Job is instantiated by the queue system and #perform is invoked to run the
+  # job.  The Job looks up an Activity that was created when the job was
+  # enqueued and calls Activity#run, passing Job#run as a block to perform the
+  # actual work. This is necessary because the Activity is designed not to care
+  # about what kind of job it's running.
+  #
+  # @see Krikri::Activity
+  # @see https://github.com/resque/resque/tree/1-x-stable
   class Job
     @queue = nil
     ##
-    # @abstract Perform the job; called by the queue system
-    def self.perform(*)
-      fail NotImplementedError
+    # Perform the job.
+    def self.perform(activity_id)
+      activity = Krikri::Activity.find(activity_id)
+      activity.run { |agent, activity_uri| run(agent, activity_uri) }
```

Comment:
I've thought about removing the `Job#run` middleman, since both existing implementations are effectively the same.  I'm sort of waiting to see what happens with enrichment queuing, but I suspect all relevant arguments want to be passed to the agent as options, yes?